### PR TITLE
[FIXED JENKINS-23411] - Fix errors in ListView::doRemoveJobFromView() 

### PR DESCRIPTION
**Pull request by oleg-nenashev**
- Throw an error if Jenkins cannot discover the item
- Synchronize delete&save operations from post requests

CID 1222626: https://scan.coverity.com/projects/94

Signed-off-by: Oleg Nenashev o.v.nenashev@gmail.com

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
It is mostly solved by @liketic in https://github.com/jenkinsci/jenkins/pull/3039. Will check if we can salvage something from this PR

File: core/src/main/java/hudson/model/ListView.java [hunk 1]
```diff
@@ -412,13 +413,15 @@ public HttpResponse doRemoveJobFromView(@QueryParameter String name) throws IOEx
...
             throw new Failure("Query parameter 'name' is required");
 
-        TopLevelItem item = resolveName(name);
-        if (item==null)
-            throw new Failure("Query parameter 'name' does not correspond to a known and readable item");
-
-        if (remove(item))
-            owner.save();
-
+        synchronized (this) {
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/ListView.java]: @liketic's PR didn't contain this sync block, which should prevent race conditions with other methods like `submit()`

File: core/src/test/java/hudson/model/ListViewTest.java [hunk 2]
```diff
@@ -64,4 +74,45 @@ public void includeRegexProgrammatic() {
         when(it.getRelativeNameFrom(any(ItemGroup.class))).thenReturn("test-item");
         assertEquals(1, view.getItems().size());
     }
+    
+    @Test
+    @Issue("JENKINS-23411")
+    @PrepareForTest({ListViewColumn.class,Items.class})
+    public void doRemoveJobFromView_missing_item() throws IOException, ServletException {
```
Comment (oleg-nenashev) [inline core/src/test/java/hudson/model/ListViewTest.java]: Probably it even works, not sure

**Comment by batmat** [comment]
Acting on my last comment. Again obviously feel free to reopen anytime if you want to get back to this. Thanks!

**Comment by oleg-nenashev** [comment]
Yes, I do not care much about it. There is a serious architecture issue in view management for Jenkins which is still around

**Pull request closed without merging**